src/axom/spin/ImplicitGrid.hpp

```diff
@@ -204,21 +204,19 @@ class ImplicitGrid
     for(int i = 0; i < NDIMS; ++i)
     {
       m_bins[i] = BinSet(m_gridRes[i]);
-      m_binData[i] = BinBitMap(&m_bins[i], BitsetType {}, 1, allocatorID);
+      m_binData[i] =
+        BinBitMap(&m_bins[i], BitsetType(numElts, allocatorID), 1, allocatorID);
```

Comment:
@kennyweiss -- now that we can call `Array::fill` on device memory, initializing `slam::Map` with a Bitset directly should Just Work™; this should fix the performance issue discussed here https://github.com/LLNL/axom/pull/742#discussion_r780523826

Reply:
Thanks Max!

Re: "should Just Work" -- awesome!
Re: performance issue from #742 -- I'll check it out once you merge this PR.

Reply:
Are you suggesting that I should re-enable initialization in `BitSet`?
(The performance issue I was seeing was on the host using a sequential policy, i.e. unrelated to device memory)

Reply:
Oh, looking at that again, I think this fix might be unrelated to your performance issue. (this does fix a segfault with device allocators though)

I think you could either use your current fix, or remove the `m_data.fill(0)` statement in the BitSet constructor, since the array should be initialized to all zeros at construction; both should work.

Reply:
Thanks -- I'll check it out.

I think the specific problem I was trying to fix was in the BitSet's copy constructor called by `ImplicitGrid::getCandidates()`,
which was setting the bits in the copy's array to zero and then copying over the data from the other array
(but I'll look closer).

i.e. in the first line of the following:
https://github.com/LLNL/axom/blob/8d841c0950a93c2d7d2e8c3006f4d59de47df395/src/axom/spin/ImplicitGrid.hpp#L338-L344

#### Context:
This came up in `Delaunay::isValid()` when there were `O(V)` queries with `O(E)` bits each (with `V` and `E` >> 1,000,000) -- so it involved tons of copies, and `findCandidates()` was taking minutes and consuming > 80% of the runtime.

Since each query only really has `O(1)` candidates, I eventually decided to modify the algorithm to only use the `ImplicitGrid` for finding the candidates and then copying this over to a `UniformGrid`.  Each of the `O(V)` queries are then directly against an array of `O(1)` integers.  And there are only `O(<num_bins>)` queries against the `ImplicitGrid`
It's a bit of a kludge, and I'd like to revisit this in the future, but it reduced the runtime of `Delaunay::isValid()` significantly.